### PR TITLE
top-level-sheet-pane sheet-transformation

### PR DESCRIPTION
**Pull request by admich**
This PR solve #769 about the wrong sheet-transformation of a top-level-sheet-pane when it start outside the screen. 

For this it  revert the changes of commit a894385 in PR #691 and find another solution to fix 
the bug of issue #499 solved by that PR.

There is also some changes in window-configuration-event.

### CODE REVIEW

File: Core/clim-basic/events.lisp [hunk 1]
```diff
@@ -241,6 +241,27 @@
...
+
+(defmethod window-configuration-event-height ((event window-configuration-event))
+  (multiple-value-bind (width height)
+        (untransform-distance  (sheet-native-transformation
+                                (sheet-parent (event-sheet event)))
+                               (window-configuration-event-native-width event)
+                               (window-configuration-event-native-height event))
+      height))
+
+
```
Comment (dkochmanski) [inline Core/clim-basic/events.lisp]: please rewrite it to `(defgeneric … (:method …))` and stick to maximum one empty line between toplevel forms

File: Core/clim-basic/mirrors.lisp [hunk 2]
```diff
@@ -176,10 +176,16 @@ very hard)."
   (let* ((parent (sheet-parent sheet))
          (sheet-region-in-native-parent
            ;; this now is the wanted sheet mirror region
-           (transform-region (sheet-native-transformation parent)
+          (if (graftp parent)
+              ;; for the TOP-LEVEL-SHEET-PANE (when the parent is a GRAFT) we don't
+              ;; clip the sheet-region with the parent-region.
```
Comment (dkochmanski) [inline Core/clim-basic/mirrors.lisp]: please add your initials and date after the comment. Also start the comment with capital letter. I.e

;; Foo goes beyond QUX so we do BAR. -- admich 2024-11-24

**Comment by admich** [comment]
@dkochmanski: I made requested changes.

**Comment by dkochmanski** [comment]
Great! thank you.

**Pull request merged**